cogs/social.py

```diff
@@ -169,12 +170,17 @@ async def lyrics(self, ctx, *, title):
         async with aiohttp.ClientSession() as cs:
             async with cs.get(f"https://some-random-api.ml/lyrics?title={title}") as r:
                 js = await r.json()
-
-                e = discord.Embed(color=discord.Color.random(), title=js['title'])
-                e.set_author(name=js['author'], url=js['links']['genius'])
-                e.description = js['lyrics']
-
-                await ctx.send(embed=e)
+                
+                lyricsArray = js['lyrics'].split("\n")
```

Comment:
This splits it for every line. If you wanted to split it between every verse you would have to do `\n\n`.
However, this is not what Flitz intended the code to do. They wanted it to split every 2000 characters (not between lines). I'll fix this issue for you because Flitz said I should fix it.

Reply:
Would splitting it every 2000 characters not often align with pre-existing letters? And wouldn't the paginator handle showing the appropriate amount of lines?

Reply:
What I meant by "(not between lines)" is that it should split between the best possible points. Preferably, it would split in between verses.

Reply:
It's the description of an embed that allowed 2048 characters, that's what the paginator uses. Is the wanted behaviour not the paginator show the maximum possible amount of lines then? In which case, wouldn't you want to implement such a feature in the paginator?

Reply:
Yes, splitting on every line isn't the best idea. I like the Duck's suggestion for splitting in between verses, but that's still not the best way to do that as verse can be 500 characters and can be 2000 characters, the best way to do this would be to pretty much find the 2000th character and then split the last newline (to make it look better instead of splitting in the middle of the word).

Reply:
If you aim for as much as possible, yeah, that's the best idea. But wouldn't it often be nicer to have 1 verse per page, considering it's music?

Reply:
> If you aim for as much as possible, yeah, that's the best idea. But wouldn't it often be nicer to have 1 verse per page, considering it's music?

This could also be done, however, I dislike this for two reasons. 
1) If the song has 30 verses the paginator would have 30 pages, which would just be annoying to go through all of them.
2) Some songs tend to have 1 or 2 line verses. It'd make the embed look pretty much empty. This is a great example: ![Image](https://cdn.dredd-bot.xyz/tAZkHZ)

Reply:
Ah, most songs I listen to tend to have more paragraph-like verses